connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignorTest.java

```diff
@@ -161,30 +161,38 @@ public void testAssignmentsWhenWorkersJoinAfterRevocations()  {
         time.sleep(assignor.delay);
         addNewEmptyWorkers("worker4");
         performStandardRebalance();
+        assertEquals(0, assignor.delay); // No revocations in this round.
         assertWorkers("worker1", "worker2", "worker3", "worker4");
         assertConnectorAllocations(0, 0, 1, 1);
         assertTaskAllocations(0, 3, 3, 3);
 
         // Fifth assignment and a fifth worker joining after a revoking rebalance.
-        // We shouldn't revoke and set a delay > initial interval
+        // We shouldn't revoke and set a delay equal to initial interval
         addNewEmptyWorkers("worker5");
         performStandardRebalance();
-        assertTrue(assignor.delay > 40);
+        assertEquals(40, assignor.delay); // First successive revoking rebalance.
```

Comment:
IMO this happens because we weren't unsetting the counter when delay goes equal to 0. This is the first successive revoking rebalance and the delay should be 40.